src/elements/MetricsSelector/MetricsSelector.js

```diff
@@ -0,0 +1,171 @@
+/*
+Copyright 2019 Iguazio Systems Ltd.
+
+Licensed under the Apache License, Version 2.0 (the "License") with
+an addition restriction as set forth herein. You may not use this
+file except in compliance with the License. You may obtain a copy of
+the License at http://www.apache.org/licenses/LICENSE-2.0.
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+implied. See the License for the specific language governing
+permissions and limitations under the License.
+
+In addition, you may not use the software for any purposes that are
+illegal under applicable law, and the grant of the foregoing license
+under the Apache 2.0 license is conditioned upon your compliance with
+such restriction.
+*/
+import React, { Fragment, useMemo, useRef, useState, useEffect, useCallback } from 'react'
+import PropTypes from 'prop-types'
+import { Field, useField } from 'react-final-form'
+import { OnChange } from 'react-final-form-listeners'
+import classNames from 'classnames'
+
+import { PopUpDialog } from 'igz-controls/components'
+import { SelectOption } from 'igz-controls/elements'
+import NameFilter from '../../common/NameFilter/NameFilter'
+import { TextTooltipTemplate, Tooltip } from 'iguazio.dashboard-react-controls/dist/components'
+import { ReactComponent as Caret } from 'igz-controls/images/dropdown.svg'
+import { generateMetricsItems } from './metricsSelector.utils'
+import { METRICS_SELECTOR_OPTIONS } from '../../types'
+
+import './metricsSelector.scss'
+
+const MetricsDropdownSelector = ({ maxSelectionNumber, metrics, name, onSelect }) => {
+  const { input } = useField(name)
+  const [nameFilter, setNameFilter] = useState('')
+  const [isOpen, setIsOpen] = useState(false)
+  const elementRef = useRef()
```

Comment:
use more specific names and not general like `element`

Reply:
+